### DISCUSSION TITLE
Using Poetry with Dagster Cloud

### DISCUSSION THREAD

**Discussion by achen187** [dagster-io/dagster#17786] [Q&A] [answered]
Whats the recommended tooling if we use Poetry for our dependency management and want to move to Dagster Cloud? The docs mention Requirements.txt and setup.py, neither of which come for free with Poetry.

**Comment by gibsondan** [comment, marked as answer]
You can generate a requirements.txt for use in Dagster Cloud from Poetry with the following command:

```
poetry export --without-hashes --format=requirements.txt > requirements.txt
```

**Reply by bnchrch** [reply]
This unfortunately doesn't handle imports of local packages.

I would recommend that people use `poetry2setup` until Dagster finds the time to support lockfiles.

**Reply by kaochikao** [reply]
@gibsondan is it in Dagster cloud's roadmap to support lockfiles?

**Comment by sairamkrish** [comment]
[poetry2setup](https://github.com/abersheeran/poetry2setup) is in archived state. `requirements.txt` alone does't satisfy dagster cloud's expectations on the code location. It is making it hard to use poetry for dagster cloud based projects.

**Comment by Nintorac** [comment]
I'm curious about the compatibility between Poetry projects and Dagster+. In your docs you mention:

> Note: pyproject.toml was introduced in [PEP-518](https://peps.python.org/pep-0518/) and meant to replace setup.py, but we may still include a setup.py for compatibility with tools that do not use this spec.

But tht doesn't really explain to me why Dagster+ doesn't support PEP518. What are the roadblocks for Dagster+ supporting PEP518?

Does Dagster+ work with uv? maybe this is the impetus I need to switch
